src/uwtools/config/formats/yaml.py

```diff
@@ -59,7 +59,9 @@ def _load(self, config_file: OptionalPath) -> dict:
         loader = self._yaml_loader
         with readable(config_file) as f:
             try:
-                return yaml.load(f.read(), Loader=loader)
+                config = yaml.load(f.read(), Loader=loader)
+                assert isinstance(config, dict)
+                return config
```

Comment:
In this case, it's actually possible for `yaml.load` to return a number of things other than a `dict`, but we expect (so far, at least) YAML that defines a top-level dictionary. So I've added an assertion to that effect, which I hope is better than lying and saying `config: dict = yaml.load(...)`, which will placate `mypy` but cause problems later when the expected `dict` object is a `list`, `int`, etc. We could alternatively raise an exception if a `dict` isn't received, but I'm not sure, at the moment, what's best. Maybe we'll see.

Reply:
Hrm, my first preference would be to enforce at load, but barring that I think I like raising an exception. However, this is already better than before knowing that